Pebble trend

jstevensog opened the pull request:
Changes to support sending PBG Trend to new version of xDrip Pebble watch face.  
Includes pngj library from maven in build.gradle.
Includes new SimpleImageEncoder.java class in UtilityModels to convert bitmaps to Pebble PNG files.
Includes Preferences.java changes to support Trend and other options.  Changes include:
- Automatically enable Broadcast Locally when Send to Pebble is turned on.
- Automatically disable Send to Pebble if Broadcast Locally is turned off.
- Automatically change values based on the selected units.  This includes High Value, Low Value, Pebble Special Value (see below), and Alert levels (well, at least the unchangeable low limit, I don't use the others).
- Hide Pebble options if Send to Pebble is NOT on.
- Change of any single Pebble setting will cause a resend to Pebble vie PebbleSync.
- Setting to Display Trend on Pebble
- Setting to Display Low Line on Pebble trend.
- Setting to Display High Line on Pebble trend.
- Setting to select Trend Period (1h, 2h, 3h, 4h).
- Setting to Display Delta.
- Setting to Display Delta Units (mmol/l or mg/dl depending on xDrip setting).
- Setting to Display Slope Arrows.
- Setting of a Special Value (default 5.5 mmol/l or 99 mg/dl, but any value can be set, so those that like 100 mg/dl are not left out).  When this value is displayed, a configurable setting will be displayed.
- Setting for the configurable special value test (default BAZINGA!).  Not everyone in the world gets the NS Easter Egg messages.

Includes changes to pref_advanced_settings.xml for the settings required above.

jstevensog committed:
Alter Dictionary to include trend image

jstevensog committed:
Alter Dictionary to include trend image

jstevensog committed:
Alter Dictionary to include trend image

jstevensog committed:
Alter Dictionary to include trend image

jstevensog committed:
Use shorter retry timer for xDrip in DexCollectionService.

jstevensog committed:
Use shorter retry timer for xDrip in DexCollectionService.

jstevensog committed:
Send unitized Delta string rather than only the delta number.  Uses full precision.

jstevensog committed:
Send unitized Delta string rather than only the delta number.  Uses full precision.

jstevensog committed:
Added PNG size debug message

jstevensog committed:
Added PNG size debug message

jstevensog committed:
Trending on Pebble.

jstevensog committed:
Trending on Pebble.

jstevensog committed:
Not Tested version of trend image transmittal.

jstevensog committed:
Not Tested version of trend image transmittal.

jstevensog committed:
Trending on Pebble.

jstevensog committed:
Trending on Pebble.

jstevensog committed:
Almost working trend generation and transfer for pebble.

Still have to sort out the watch side before I test more.
Currently, trending is not being sent out, as line 350 has a /* at the
start.

jstevensog committed:
Almost working trend generation and transfer for pebble.

Still have to sort out the watch side before I test more.
Currently, trending is not being sent out, as line 350 has a /* at the
start.

jstevensog committed:
Added app.iml

jstevensog committed:
Added app.iml

jstevensog committed:
Working data transfer.

Looking at why the pebble can't convert the PNG.

jstevensog committed:
Working data transfer.

Looking at why the pebble can't convert the PNG.

jstevensog committed:
Minor fixes to the trend sending code.

Make sure the last chunk is the correct size.

jstevensog committed:
Minor fixes to the trend sending code.

Make sure the last chunk is the correct size.

jstevensog committed:
Included SimpeImageEncoder.java to create 8bit PNG images.  Not sure it is done right yet.  That meant adding references to ar.com.hjg.pngj library to app.iml and build.gradle.

jstevensog committed:
Included SimpeImageEncoder.java to create 8bit PNG images.  Not sure it is done right yet.  That meant adding references to ar.com.hjg.pngj library to app.iml and build.gradle.

jstevensog committed:
Modified SimpleImageEncoder and BgSparklineBuilder.

Saving for now while I try a fresh go at SimpleImageEncoder.

jstevensog committed:
Modified SimpleImageEncoder and BgSparklineBuilder.

Saving for now while I try a fresh go at SimpleImageEncoder.

jstevensog committed:
Pebble Trend in PebbleSync working.

Modified SimpleImageEncoder to work.  Ensured that only indexed PNG is
produced (image type 3), with the transparent index set.
Also removed quantisation when using a transparent background, as for some
reason it called half the background colours slightly less black.
Still working on why red sometimes appears as black on Pebble.

jstevensog committed:
Pebble Trend in PebbleSync working.

Modified SimpleImageEncoder to work.  Ensured that only indexed PNG is
produced (image type 3), with the transparent index set.
Also removed quantisation when using a transparent background, as for some
reason it called half the background colours slightly less black.
Still working on why red sometimes appears as black on Pebble.

jstevensog committed:
Multiple Changes

Added Preferences for Pebble special value and text (to replace BAZINGA!).
Added automation to change of units so that High and Low values and Pebble
Special Value are
altered appropriate to the units selected.
Added automatically setting Broadcast Locally on if Send to Pebble is
turned on.
Added automatically setting Send to Pebble off if Broadcast Locally is
turned off.

jstevensog committed:
Multiple Changes

Added Preferences for Pebble special value and text (to replace BAZINGA!).
Added automation to change of units so that High and Low values and Pebble
Special Value are
altered appropriate to the units selected.
Added automatically setting Broadcast Locally on if Send to Pebble is
turned on.
Added automatically setting Send to Pebble off if Broadcast Locally is
turned off.

jstevensog committed:
Latest PebbleSync changes.

Allows setting the BAZINGA! message and level in xDip.
Also added support for Vibration Level to pebble from xDrip.  Not yet
being used.

jstevensog committed:
Latest PebbleSync changes.

Allows setting the BAZINGA! message and level in xDip.
Also added support for Vibration Level to pebble from xDrip.  Not yet
being used.

jstevensog committed:
Improved No Sensor Code in PebbleSync.java

jstevensog committed:
Improved No Sensor Code in PebbleSync.java

jstevensog committed:
Removed Merge Conflicts

jstevensog committed:
Removed Merge Conflicts

jstevensog committed:
Remove DexCollectionService merge conflict

jstevensog committed:
Remove DexCollectionService merge conflict

jstevensog committed:
Fix build.gradle after merge conflic resolution.

Created an error in the file that had to be fixed.

jstevensog committed:
Fix build.gradle after merge conflic resolution.

Created an error in the file that had to be fixed.

jstevensog committed:
Yet another app.iml merge conflict fix.

jstevensog committed:
Yet another app.iml merge conflict fix.

jstevensog committed:
Added more icons logic.

Modify BgReading() to return ?RF if a packets are missed, and
?SN if sensor is not active.

jstevensog committed:
Added more icons logic.

Modify BgReading() to return ?RF if a packets are missed, and
?SN if sensor is not active.

jstevensog committed:
Align with current master.

Fixed merge conflicts in utils\Preferences.
Removed additional creation of BgGraphBuilder instance in PebbleSync.bgDelta.

jstevensog committed:
Align with current master.

Fixed merge conflicts in utils\Preferences.
Removed additional creation of BgGraphBuilder instance in PebbleSync.bgDelta.

jstevensog committed:
Fix bug in sendData.

Was not incrementing sendStep after call to sendDictionary.  Not sure ho
it got in there, but fixed and working well now.
May have failed to check out and commit from test branch.

jstevensog committed:
Fix bug in sendData.

Was not incrementing sendStep after call to sendDictionary.  Not sure ho
it got in there, but fixed and working well now.
May have failed to check out and commit from test branch.

jstevensog committed:
Added Delta preference.

To turn on and off display of Delta.  Does not impact display of Message.

jstevensog committed:
Added Delta preference.

To turn on and off display of Delta.  Does not impact display of Message.

jstevensog committed:
Add Display Delta/Arrows preferences and code.

Added preferences and code to allow users to chose to display delta and
arrows from within xDrip.

jstevensog committed:
Add Display Delta/Arrows preferences and code.

Added preferences and code to allow users to chose to display delta and
arrows from within xDrip.

jstevensog committed:
Fixed null mBgReading but, and cleand up code.

Was not testing for a null mBgReading object which happens when sensor is
stopped.  This crashes xDrip.

Set a single no_signal boolean for use throughout the class, that is set
once IF mBgReading is NOT null.  Allows for correct indication/return of
values when a packet has not been captured.

TODO:  Need to make the xBridge Battery return some form of indication
that this figure may be stale.  Right now it stays at the last value.  Not
sure that setting this back to 0 is the correct way to deal with it.

jstevensog committed:
Fixed null mBgReading but, and cleand up code.

Was not testing for a null mBgReading object which happens when sensor is
stopped.  This crashes xDrip.

Set a single no_signal boolean for use throughout the class, that is set
once IF mBgReading is NOT null.  Allows for correct indication/return of
values when a packet has not been captured.

TODO:  Need to make the xBridge Battery return some form of indication
that this figure may be stale.  Right now it stays at the last value.  Not
sure that setting this back to 0 is the correct way to deal with it.

jstevensog committed:
Cleanup merge conflicts with master.

jstevensog committed:
Cleanup merge conflicts with master.

jstevensog committed:
Added MultiDex support and fixed bugs

Required MultiDex support as we had run out of Dex indices and PebbleTrend
branch would not build.
Fixed bugs in Preferences, where some Pebble options didn't disappear when
turning of Pebbly Sync.

jstevensog committed:
Added MultiDex support and fixed bugs

Required MultiDex support as we had run out of Dex indices and PebbleTrend
branch would not build.
Fixed bugs in Preferences, where some Pebble options didn't disappear when
turning of Pebbly Sync.

jstevensog committed:
Draw transparent High/Low lines

jstevensog committed:
Draw transparent High/Low lines

jstevensog committed:
Matched AdrianLXM's "Watch Integraton" preferences.

As well as I could.

jstevensog committed:
Matched AdrianLXM's "Watch Integraton" preferences.

As well as I could.

jstevensog committed:
Fixed all settings issues.

Still need to figure out how to clear the trend from the watch if the Send
Trend preference is turned off.  Right now it is a manual restart of the
app/face.

jstevensog committed:
Fixed all settings issues.

Still need to figure out how to clear the trend from the watch if the Send
Trend preference is turned off.  Right now it is a manual restart of the
app/face.

jstevensog committed:
Code Cleanup prior to merge.

jstevensog committed:
Code Cleanup prior to merge.

AdrianLxM commented:
Hi, I think you are mixing up settings. `broadcast_data_through_intents` will send data to other Android Apps via a local broadcast. Those other Apps were NitghtWatch and still is HAPP e.g. and is independent of showing data on the pebble.

https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256/files#diff-160e4330ff39df6125d6a07d85107ba3R535
... looks like if I want to deactivate sending Data to the Artificial-Pancreas-App it will automatically activate pebble sync - no matter if I have a pebble or not.

https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256/files#diff-160e4330ff39df6125d6a07d85107ba3R445
... looks like it will start sending data to the Artificial-Pancreas-App if pebble integration is activated.

I think the pebble integration should only depend on `broadcast_to_pebble` (and the actual sync service only does, it seems to be just mixed up in the settings).

---

And one other thing I'm not sure about is the build/dependency part:
You added a `pngj-2.1.0.jar` directly to the xdrip repository. From the build.gradle it looks like it would be downloaded and compiled from a maven repository.

jstevensog commented:
Hi Adrian,
Broadcast locally has always been a pre-requisite for sending to Pebble.  All I have done is make Broadcast locally automatically turn on if Send to Pebble is on, and torn send to Pebble off if Broadcast locally is turned off.  I I have that logic wrong, I am happy to fix it, but my testing has shown this is what happens.

I am not sure how the pngj library got there, I will remove it and let it be fetched from maven.
It was late when I did this, so missed that and the wear/wear.iml.  
Cheers

---- AdrianLxM wrote ----

> Hi, I think you are mixing up settings. broadcast_data_through_intents will send data to other Android Apps via a local broadcast. Those other Apps were NitghtWatch and still is HAPP e.g. and is independent of showing data on the pebble.
> 
> https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256/files#diff-160e4330ff39df6125d6a07d85107ba3R535
> ... looks like if I want to deactivate sending Data to the Artificial-Pancreas-App it will automatically activate pebble sync - no matter if I have a pebble or not.
> 
> https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256/files#diff-160e4330ff39df6125d6a07d85107ba3R445
> ... looks like it will start sending data to the Artificial-Pancreas-App if pebble integration is activated.
> 
> I think the pebble integration should only depend on broadcast_to_pebble (and the actual sync service only does).
> 
> And one other thing I'm not sure about is the build/dependency part:
> You added a pngj-2.1.0.jar directly to the xdrip repository. From the build.gradle it looks like it would be downloaded and compiled from a maven repository.
> 
> —
> Reply to this email directly or view it on GitHub.￼

AdrianLxM commented:
When I tested it for a friend the local broadcast was not necessary.
And if it was, I would consider it a bug.

`broadcast_data_through_intents` is just queried in BgSendQueue and the two if blocks that handle the local broadcast and the pebble sync are disjoint. I looked back a bit and they were disjoint in May 2015 already... didn't look further back.

jstevensog commented:
Well, if it is a bug, I am happy to remove it.  I have answered quite a few
support requests, and turning on broadcast_data_through_intents was what
fixed it.  These cases would always have been using the last xDrip release
before we started sending out the betas.

I will try and find time today to run a test, then update the branch.
Cheers

On Wed, Jan 20, 2016 at 6:47 AM, AdrianLxM notifications@github.com wrote:

> When I tested it for a friend the local broadcast was not necessary.
> And if it was, I would consider it a bug.
> 
> broadcast_data_through_intents is just queried in BgSendQueue and the two
> if blocks that handle the local broadcast and the pebble sync are disjoint.
> I looked back a bit and they were disjoint in May 2015 already... didn't
> look further back.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256#issuecomment-172965101
> .

## 

John Stevens
"You are how you live, not what you have."

jstevensog committed:
Removed pngj library that I accidentally included.

Originally I had placed it in the app\libs directory, but later changed
this to maven repositoriy in build.grade.  Thanks to Adrian Mittermayr for
pointing this out.

jstevensog committed:
Removed pngj library that I accidentally included.

Originally I had placed it in the app\libs directory, but later changed
this to maven repositoriy in build.grade.  Thanks to Adrian Mittermayr for
pointing this out.

jstevensog committed:
Removed interlink in Preferences.java

The dependency of Send to Pebble on Broadcast Locally is no longer an
issue.  Removed all interlinking code between these two settings.

jstevensog committed:
Removed interlink in Preferences.java

The dependency of Send to Pebble on Broadcast Locally is no longer an
issue.  Removed all interlinking code between these two settings.

jstevensog commented:
OK, I have tested the independence of Send To Pebble and Broadcast Locally
settings and there is no issue as Adrian pointed out.
I have removed all the interlinking code in Preferences.java, and also
removed the pngj library file I mistakenly put it.
PR should now be a lot easier to merge, not that there were any conflicts.

I wouldn't want to merge this right away, until I get some indication from
Pebble on when they will fix the aplite issues I have found out about.
But once that is resolved, and I have built and tested the aplite face, I
would like to merge then.  Until then, this should NOT be in a beta release.
Cheers

On Wed, Jan 20, 2016 at 7:50 AM, John Stevens jstevensog@gmail.com wrote:

> Well, if it is a bug, I am happy to remove it.  I have answered quite a
> few support requests, and turning on broadcast_data_through_intents was
> what fixed it.  These cases would always have been using the last xDrip
> release before we started sending out the betas.
> 
> I will try and find time today to run a test, then update the branch.
> Cheers
> 
> On Wed, Jan 20, 2016 at 6:47 AM, AdrianLxM notifications@github.com
> wrote:
> 
> > When I tested it for a friend the local broadcast was not necessary.
> > And if it was, I would consider it a bug.
> > 
> > broadcast_data_through_intents is just queried in BgSendQueue and the
> > two if blocks that handle the local broadcast and the pebble sync are
> > disjoint. I looked back a bit and they were disjoint in May 2015 already...
> > didn't look further back.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256#issuecomment-172965101
> > .
> 
> ## 
> 
> John Stevens
> "You are how you live, not what you have."

## 

John Stevens
"You are how you live, not what you have."

jstevensog committed:
Fixed bug in sendTrend.

When Display Trend setting was OFF, the TREND_BEGIN dictionary was still
sent.  This prevents that occurring.
Also cleaned up some commented out code.

jstevensog committed:
Fixed bug in sendTrend.

When Display Trend setting was OFF, the TREND_BEGIN dictionary was still
sent.  This prevents that occurring.
Also cleaned up some commented out code.

jstevensog committed:
Merged Preferences from testmerge3

jstevensog committed:
Merged Preferences from testmerge3

jstevensog committed:
Rebased to master.

jstevensog committed:
Rebased to master.

jstevensog committed:
Can't remember what I changed.  Likely the new watch dictionary stuff.

jstevensog committed:
Can't remember what I changed.  Likely the new watch dictionary stuff.

jstevensog committed:
Attempt to fix lollipop downtime issue.
Same change as the DowntimeFix branch

jstevensog committed:
Attempt to fix lollipop downtime issue.
Same change as the DowntimeFix branch

jstevensog committed:
Merge missing preferences in PebbleTrend

jstevensog committed:
Merge missing preferences in PebbleTrend

jstevensog commented:
Latest PebbleTrend Branch PR.
Please review the bg_missed\* preferences in pref_notifications.xml and Preferences.java.
These seem to have gone missing from master, and I am not sure why.  Mind you, it has been some time since this was first raised, and it has gone through a couple of rebases.
Cheers

AdrianLxM commented on app/build.gradle:
```diff
@@ -117,10 +121,18 @@ dependencies {
+    compile 'com.squareup.okhttp:okhttp:2.5.0'
+    compile 'com.google.code.gson:gson:2.3'
+    compile 'com.squareup.retrofit:converter-gson:2.0.0-beta2'
+    compile 'com.squareup.retrofit:retrofit:2.0.0-beta2'
```
It looks like there are some duplicates here.

tzachi-dar replied:
yes, 
com.squareup.okhttp:mockwebserver,
com.nispok:snackbar,
com.android.support:appcompat,
com.squareup.okhttp:okhttp:2.5.0, 
com.google.code.gson:gson:2.3,
com.squareup.retrofit:converter-gson
com.squareup.retrofit:retrofit

Are all duplicate.

You only need to leave ar.com.hjg:pngj :-)

AdrianLxM commented:
@tzachi-dar deleted this preference when he wrote a separate Activity for that.
It was in this commit: https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/commit/5c1e87e6baaca0b7068ff59f42b583eb8a0eca96#diff-160e4330ff39df6125d6a07d85107ba3

AdrianLxM commented on app/src/main/res/xml/pref_notifications.xml:
```diff
@@ -115,6 +115,21 @@ S        <PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/and
             </PreferenceCategory>
 
             <PreferenceCategory
+                android:title="Missed Readings">
```
I guess this would double the missed readings settings as it is already there in line 95:
https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256/files#diff-9abb79954cd4af4a8b02a7e89d92964bR95

tzachi-dar replied:
Yes, remove this lines all together.

tzachi-dar commented on app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java:
```diff
@@ -373,7 +373,7 @@ private void ArmTimer(Context ctx) {
             alarm.cancel(wakeIntent);
         wakeIntent = PendingIntent.getService(this, 0, new Intent(this, this.getClass()), 0);
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.LOLLIPOP) {
```
Are you sure  alarm.setAlarmClock should not be used in lollipop?

jstevensog replied:
Hi All,
As we are not so certain of how to fix the Downtime issue, and the change
to Notifications.java didn't help in any way, I have backed out this change
in PebbleSync branch.
Should be fine to merge, though I might see if I can get the watch face
install in as well.  Might have some time today to finish that.
Cheers

On Thu, Jun 16, 2016 at 1:59 AM, tzachi-dar notifications@github.com
wrote:

> In
> app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java
> https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256#discussion_r67191138
> :
> 
> > @@ -373,7 +373,7 @@ private void ArmTimer(Context ctx) {
> >          if (wakeIntent != null)
> >              alarm.cancel(wakeIntent);
> >          wakeIntent = PendingIntent.getService(this, 0, new Intent(this, this.getClass()), 0);
> > -        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
> > -        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.LOLLIPOP) {
> 
> Are you sure alarm.setAlarmClock should not be used in lollipop?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256/files/a063e56a27b0e35ee4f36ddb0ed7af3f50a9d3a7#r67191138,
> or mute the thread
> https://github.com/notifications/unsubscribe/AIQs80SPnDVLmoNpZLXSexTYp8XVLXH_ks5qMCFwgaJpZM4HHlcy
> .

## 

John Stevens
"You are how you live, not what you have."

tzachi-dar commented on app/src/main/res/xml/pref_data_source.xml:
```diff
@@ -17,6 +17,16 @@
             android:key="share_key"
             android:shouldDisableView="true"
             android:defaultValue="SM00000000"/>
+        <EditTextPreference
```
This two are already in the file, no need to have them twice.

tzachi-dar commented on wear/wear.iml:
```diff
@@ -30,6 +31,7 @@
     <output-test url="file://$MODULE_DIR$/build/intermediates/classes/androidTest/xdrip/debug" />
     <exclude-output />
     <content url="file://$MODULE_DIR$">
+<<<<<<< HEAD:wear/wear.iml
```
There are merge conflicts here, but I don't think you need to submit this file in any case.

jstevensog replied:
OK, guys.  Thanks so much for your feedback.
I have checked out wear/wear.iml, app/build.gradle, and
app/src/main/res/xml/pref_notifications.xml from master to remove those
conflicts.  I have modified utils/Preferences.java to remove the null
pointer exceptions I got a couple of days ago.

I have left the AlarmClock check in Notifications.java for now.  It hasn't
solved the down time issue at all, but I thought it prudent to leave it
there until we have a better idea.
Will do a build and test run and update the PR.
Thanks again and Cheers

On Thu, Jun 16, 2016 at 2:13 AM, tzachi-dar notifications@github.com
wrote:

> In wear/wear.iml
> https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256#discussion_r67193882
> :
> 
> > @@ -30,6 +31,7 @@
> >      <output-test url="file://$MODULE_DIR$/build/intermediates/classes/androidTest/xdrip/debug" />
> >      <exclude-output />
> >      <content url="file://$MODULE_DIR$">
> > +<<<<<<< HEAD:wear/wear.iml
> 
> There are merge conflicts here, but I don't think you need to submit this
> file in any case.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256/files/a063e56a27b0e35ee4f36ddb0ed7af3f50a9d3a7#r67193882,
> or mute the thread
> https://github.com/notifications/unsubscribe/AIQs8zWRScxJE-hQexij5j0-QXkksUSdks5qMCSngaJpZM4HHlcy
> .

## 

John Stevens
"You are how you live, not what you have."

jstevensog replied:
OK, I have hit a snag.  wear/wear.iml needs for some reason to be here.  If
I git rm --cached, it deletes it from the branch, which I believe will
delete it from master if this is merged.  Is that the case?
I really don't want to do that.  I will check it out from master again
prior to committing and pushing PebbleTrend, to remove the merge
conflicts.  Once it is in, I will make sure I never add an iml file to my
commits.
Cheers

On Thu, Jun 16, 2016 at 7:18 AM, John Stevens jstevensog@gmail.com wrote:

> OK, guys.  Thanks so much for your feedback.
> I have checked out wear/wear.iml, app/build.gradle, and
> app/src/main/res/xml/pref_notifications.xml from master to remove those
> conflicts.  I have modified utils/Preferences.java to remove the null
> pointer exceptions I got a couple of days ago.
> 
> I have left the AlarmClock check in Notifications.java for now.  It hasn't
> solved the down time issue at all, but I thought it prudent to leave it
> there until we have a better idea.
> Will do a build and test run and update the PR.
> Thanks again and Cheers
> 
> On Thu, Jun 16, 2016 at 2:13 AM, tzachi-dar notifications@github.com
> wrote:
> 
> > In wear/wear.iml
> > https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256#discussion_r67193882
> > :
> > 
> > > @@ -30,6 +31,7 @@
> > >      <output-test url="file://$MODULE_DIR$/build/intermediates/classes/androidTest/xdrip/debug" />
> > >      <exclude-output />
> > >      <content url="file://$MODULE_DIR$">
> > > +<<<<<<< HEAD:wear/wear.iml
> > 
> > There are merge conflicts here, but I don't think you need to submit this
> > file in any case.
> > 
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/256/files/a063e56a27b0e35ee4f36ddb0ed7af3f50a9d3a7#r67193882,
> > or mute the thread
> > https://github.com/notifications/unsubscribe/AIQs8zWRScxJE-hQexij5j0-QXkksUSdks5qMCSngaJpZM4HHlcy
> > .
> 
> ## 
> 
> John Stevens
> "You are how you live, not what you have."

## 

John Stevens
"You are how you live, not what you have."

jstevensog committed:
Remove redundancy and merge conflicts.

* Remove redundant settings in pref_notifications.xml.
* Remove prefereces in Preferences.java that caused NP errors.
* Checked out wear/wear.iml from master to avoid merge conflicts.
* Removed duplicates in app\build.gradle compile statements.
* Tested, so far so good.

jstevensog committed:
Remove redundancy and merge conflicts.

* Remove redundant settings in pref_notifications.xml.
* Remove prefereces in Preferences.java that caused NP errors.
* Checked out wear/wear.iml from master to avoid merge conflicts.
* Removed duplicates in app\build.gradle compile statements.
* Tested, so far so good.

jstevensog committed:
Back out Notifications change.

jstevensog committed:
Back out Notifications change.

jstevensog committed:
Fix Trending for Pebble Classic.

Included fixes from jamorham's xDrip+.
* Fixed SimpleImageEncoder for monochrome images.
* Created separate call to BgSparklineBuilder in PebbleSync for
monochrome images.  This matches the watch face trend area size.

jstevensog committed:
Fix Trending for Pebble Classic.

Included fixes from jamorham's xDrip+.
* Fixed SimpleImageEncoder for monochrome images.
* Created separate call to BgSparklineBuilder in PebbleSync for
monochrome images.  This matches the watch face trend area size.

jstevensog committed:
Merge branch 'PebbleTrend' of https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental into PebbleTrend

jstevensog committed:
Merge branch 'master' into PebbleTrend

jstevensog committed:
Added xDrip-pebble2.pbw to assets.

Still to do:
* Code up the watch face installation.

jstevensog committed:
Changed UUID of Watch Face.

This will allow xDrip and xDrip-Trend to co-exist on the Pebble Store.

jstevensog committed:
Added new watch face build.

Still not installing from app, but keeping things up to date.

Commit:
Merge branch 'PebbleTrend' into adrian-pebbletrend

Conflicts:
	app/build.gradle

Commit:
pebble config bundled

Commit:
pebble special value numeric

jstevensog committed:
Merge pull request #376 from StephenBlackWasAlreadyTaken/adrian-pebbletrend

Adrian pebbletrend

Commit:
Remove duplicate settings

They seem to have been duplicated during the merges.

jstevensog committed:
Merge pull request #377 from StephenBlackWasAlreadyTaken/PebbleTrend-settingsPatch

Remove duplicate settings

jstevensog committed:
Updated xDrip-pebble2.pbw

This is the new Classic Watch face by AdrianLXM.
Also has the new UUID, which I appear to have forgotten to put in earlier.

jstevensog committed:
xDrip-pebble2 Version Update.

Forgot to update it in the appinfo.json.  This is the version in the
Pebble store.

Commit:
q

Commit:
undo bad edit

Commit:
alert settings for pebble

Commit:
typo

Commit:
remove vibe key after it is sent

Commit:
remove vibe key again

Commit:
show delta and arrow by default

jstevensog committed:
Merge pull request #379 from StephenBlackWasAlreadyTaken/pebbletrend-alarmsettings

Pebbletrend alarmsettings

jstevensog committed:
Updated xDrip-pebble2.pbw

That is all.
This version does not re-raise the NO BLUETOOTH vibration alert.

jstevensog committed:
Improved Watch Face

Updated watch face.  Now includes some settings in Pebble app.
Notes:
* Time can display seconds - Tested.
* No Bluetooth vibration can either repeat every 15 15 minutes, or not.
* Foreground and Background colour selection does not yet work.

jstevensog committed:
Updated Watch Face and Auto Install Code.

Last commit watch face had serious issues.  This commit watch face fixes
all of those.
Also corrected the Auto Install code to prevent installation attempts if a
watch is out of range or not connected (Thanks AdrianLXM for pointing this
out).

jstevensog committed:
Updated watch face.

Settings now will not show foreground and background colour settings
unless a colour capable watch is connected.

jstevensog committed:
Updated pebble watch face.

Altered all monochrome indicator images to be white on black.
Looks much better with AdrianLXM's changes.

jstevensog committed:
Updated Pebble watch face.  Backlight.

Updated the watch face to add a setting to keep the back light on whilst
the Pebble is connected to power for charging.
This will increase time to charge, but allow the watch to be used as a
glanceable display at night.
Requested by AdrianLXM.

jstevensog committed:
Removed debug from Pebble watch face.

jstevensog committed:
Updated Pebble watch face.

 * Fixed data losing month on day change.
 * Fixed date chopping p and g character tails.

jstevensog committed:
Updated watch face.

Removed debug and brought it up to latest.  Not sure why, but it appears I
didn't put the latest version in the last commit.
